solr/core/src/test/org/apache/solr/handler/admin/PropertiesRequestHandlerTest.java

```diff
@@ -62,13 +61,9 @@ public void testDisabledRedaction() throws Exception {
 
   @SuppressWarnings({"unchecked"})
   private NamedList<Object> readProperties() throws Exception {
-    String xml = h.query(req(
-        CommonParams.QT, "/admin/properties",
-        CommonParams.WT, "xml"
-    ));
-
-    XMLResponseParser parser = new XMLResponseParser();
-    return (NamedList<Object>)
-        parser.processResponse(new StringReader(xml)).get("system.properties");
+    SolrQueryResponse rsp = new SolrQueryResponse();
+    SolrQueryRequest req = req();
+    new PropertiesRequestHandler().handleRequestBody(req, rsp);
```

Comment:
If we wanted to test the properties handler within the core container then

```suggestion
    h.getCoreContainer().getInfoHandler().getPropertiesHandler().handleRequestBody(req, rsp);
```

could be an alternative here.

The `qt=/admin/properties` previously would indirectly also test accessibility via that path, though I'm not sure how much that is a key part of the test? Or if there is a replacement -- `/admin/info/properties` (?) -- and/or how feasible using that in the test would be ...

What do you think?

Reply:
Please use `new EmbeddedSolrServer(h.getCore());` to give you a client to make a normal request into Solr.  Ideally our test infrastructure would make this trick/technique more obvious / transparent.